### PR TITLE
Feedback box

### PR DESCRIPTION
**Pull request by LisaJD**

### CODE REVIEW

File: eahub/base/static/global/styles/components/_feedback.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
+@media (min-width: 601px) {
+#feedback {
```
Comment (sebgrebe) [inline eahub/base/static/global/styles/components/_feedback.scss:2]: minor issue: the indentations seem to have been messed up in this file

File: eahub/templates/components/feedback.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              <button id="submit2" style="width: auto;" onClick="refreshFeedback()" class="btn submitbtn">Submit more feedback</button>
+              </div>
+          </div>     
+
+        </div>
+    </div>  
+  </div>
+
+
+  <script>
```
Comment (sebgrebe) [inline eahub/templates/components/feedback.html:54]: This should be in a separate js file, ideally.

Even better, we would use the bootstrap modal functionality here, which would save us quite a bit of JS: https://getbootstrap.com/docs/5.0/components/modal/
(If you feel this is too much work, feel free to leave this out, and create a refactoring ticket for this to be tackled later)

File: eahub/base/static/global/styles/components/_feedback.scss [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+  .submitbtn {
+  color: #fff;
+  background: rgb(99, 82, 116);
+  border-color: rgb(99, 82, 116);
+  width: 30%;
+}
+
+  .feedbackbtn, .feedbackbtn-mobile {
```
Comment (sebgrebe) [inline eahub/base/static/global/styles/components/_feedback.scss:101]: On mobile, the button scrolls out of view if I scroll down

File: eahub/templates/eahub/profiles.html [hunk 4]
```diff
@@ -52,6 +52,10 @@
 
 {% endblock %}
 
+{% block feedback %}
```
Comment (sebgrebe) [inline eahub/templates/eahub/profiles.html:55]: I couldn't see the button. When I moved the feedback block into the above block I could see it

**Review by sebgrebe — CHANGES_REQUESTED** [review]

File: eahub/templates/components/feedback.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      body: new URLSearchParams(new FormData(form)),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).then(response => {
+      if(response.status == "200") {
+        document.getElementById("successmsg").style.display = "block";
+        document.getElementById("form").style.display = "none"; 
+      }
+      if(response.status == "418") {
```
Comment (viktor-yunenko) [inline eahub/templates/components/feedback.html:77]: I recommend to use `if respnse.status !== "200"` instead

File: eahub/feedback/views.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+def ajax_post_view(request):
+    model = Feedback
+    form_class = FeedbackForm
+    if request.method == "POST":
+        form = FeedbackForm(request.POST)
+        if form.is_valid():
+            form.save()
+            return HttpResponse(status=200)
+        else:
+            return HttpResponse(status=418)
```
Comment (viktor-yunenko) [inline eahub/feedback/views.py:16]: would be great to use CreateView instead for simplicity sake

File: eahub/feedback/forms.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+class FeedbackForm(forms.ModelForm):
+    class Meta:
+        model = Feedback
+        fields = ("message", "email", "page_url")
+        widgets = {
+            "message": forms.Textarea(
+                attrs={"rows": 3, "maxlength": 5000, "required": "true"}
+            )
+        }
+        labels = {"message": ("Feedback"), "email": ("Email address (optional)")}
```
Comment (viktor-yunenko) [inline eahub/feedback/forms.py:14]: if we go with CreateView I don't think this file would be needed

File: eahub/feedback/views.py (hunk 6 above)
Comment (sebgrebe) [reply eahub/feedback/views.py:16]: please add ToDo statement for refactoring

File: eahub/feedback/forms.py (hunk 7 above)
Comment (sebgrebe) [reply eahub/feedback/forms.py:14]: please add ToDo statement for refactoring

**Comment by sebgrebe** [comment]
You can use bootstrap5 modal: https://getbootstrap.com/docs/5.0/components/modal/

**Pull request closed without merging**